### PR TITLE
added Bone.rotate and Bone.scale

### PR DESCRIPTION
**Pull request by abow**
Here is a PG I used for testing:

http://www.babylonjs-playground.com/#I6RJJ#6
http://www.babylonjs-playground.com/#I6RJJ#7
http://www.babylonjs-playground.com/#I6RJJ#8
http://www.babylonjs-playground.com/#I6RJJ#9

### CODE REVIEW

File: src/Math/babylon.math.ts [hunk 1]
```diff
@@ -1074,9 +1074,12 @@
         }
 
         public static TransformNormalToRef(vector: Vector3, transformation: Matrix, result: Vector3): void {
-            result.x = (vector.x * transformation.m[0]) + (vector.y * transformation.m[4]) + (vector.z * transformation.m[8]);
-            result.y = (vector.x * transformation.m[1]) + (vector.y * transformation.m[5]) + (vector.z * transformation.m[9]);
-            result.z = (vector.x * transformation.m[2]) + (vector.y * transformation.m[6]) + (vector.z * transformation.m[10]);
+            var x = (vector.x * transformation.m[0]) + (vector.y * transformation.m[4]) + (vector.z * transformation.m[8]);
+            var y = (vector.x * transformation.m[1]) + (vector.y * transformation.m[5]) + (vector.z * transformation.m[9]);
+            var z = (vector.x * transformation.m[2]) + (vector.y * transformation.m[6]) + (vector.z * transformation.m[10]);
+            result.x = x;
```
Comment (wyklion) [inline src/Math/babylon.math.ts]: what's the different?

**Comment by abow** [comment]
This version can be used to transform a normal to itself.

File: src/Math/babylon.math.ts (hunk 1 above)
Comment (abow) [inline src/Math/babylon.math.ts]: This version can be used to transform a normal to itself.

**Comment by deltakosh** [comment]
Excellent! Do you mwant je updating the doc as well? http://doc.babylonjs.com/tutorials/How_to_use_Bones_and_Skeletons

**Comment by abow** [comment]
No problem.  I'll do that in a few days.

**Comment by abow** [comment]
I got this working while an animation is playing:

http://www.babylonjs-playground.com/#11BH6Z#177

For some reason, scaling without rotating the bone doesn't work.

http://www.babylonjs-playground.com/#11BH6Z#178

If I animate the scaling it shows some erratic movement:

http://www.babylonjs-playground.com/#11BH6Z#179

Do you know what could be causing that?

**Pull request merged**